src/impi/core.cljs

```diff
@@ -227,6 +260,25 @@
        (map #(build! index attr %))
        (replace-children container)))
 
+; Does not cache individual shapes as any change requires a full redraw
+(defmethod update-prop! :pixi.graphics/shapes 
+  [graphics-obj _ _ shapes]
+  (.clear graphics-obj)
+  (doseq [{fill       :pixi.shape/fill
```

Comment:
Alignment is off a little here. You could also use `:keys` for brevity (it's a little known functionality that `:keys` works with namespaced keywords as well).

```clojure
(doseq [{:keys [:pixi.shape/fill :pixi.shape/line] :as shape}
        (if (map? shapes) (vals shapes) shapes)]
  ...)
```

Though perhaps it would be clearer with a `let`?

```clojure
(doseq [shape (if (map? shapes) (vals shapes) shapes)]
  (let [{:keys [:pixi.shape/fill :pixi.shape/line]} shape]
    ...))
```